src/main/kotlin/org/jetbrains/research/libsl/visitors/ExpressionVisitor.kt

```diff
@@ -211,83 +224,49 @@ class ExpressionVisitor(
+            num = num.dropLast(1)
             return IntegerLiteral(
-                ctx.text.toInt(),
-                null,
-                posGetter.getCtxPosition(context.fileName, ctx))
-        } else {
-            return when(ctx.suffix().text) {
-                "b" -> IntegerLiteral(
-                    ctx.text.dropLast(1).toByte(),
-                    "b",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "ub" -> UnsignedInt8Literal(
-                    ctx.text.dropLast(2).toUByte(),
-                    "ub",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "s" -> IntegerLiteral(
-                    ctx.text.dropLast(1).toShort(),
-                    "s",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "us" -> UnsignedInt16Literal(
-                    ctx.text.dropLast(2).toUShort(),
-                    "us",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                null -> IntegerLiteral(
-                    ctx.text.toInt(),
-                    "",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "u" -> UnsignedInt32Literal(
-                    ctx.text.dropLast(1).toUInt(),
-                    "u",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "L" -> IntegerLiteral(
-                    ctx.text.dropLast(1).toLong(),
-                    "L",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "uL" -> UnsignedInt64Literal(
-                    ctx.text.dropLast(2).toULong(),
-                    "uL",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                else -> throw IllegalArgumentException("Incorrect integer suffix")
-            }
+                convertBinHexOctToPrimitives(num, "l"),
+                "L",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
         }
+        return IntegerLiteral(
+            convertBinHexOctToPrimitives(num, "i"),
+            null,
+            posGetter.getCtxPosition(context.fileName, ctx)
+        )
     }
 
+    fun convertBinHexOctToPrimitives(num: String, type: String): Number {
+        val result: Number = if (num.startsWith(HEX_PREFIX)) {
+            if ("i".equals(type)) parseInt(num.drop(2), 16) else parseLong(num.drop(2), 16)
+        } else if (num.startsWith(BIN_PREFIX)) {
+            if ("i".equals(type)) parseInt(num.drop(2), 2) else parseLong(num.drop(2), 2)
+        } else if (num.startsWith(OCT_PREFIX) && num.length > 1) {
+            if ("i".equals(type)) parseInt(num.drop(1), 8) else parseLong(num.drop(1), 8)
+        } else {
+            if ("i".equals(type)) num.toInt() else num.toLong()
```

Comment:
Seems to me that it is possible to make a local variable for `"i".equals(type)`

Reply:
Fixed